### PR TITLE
gh-123228: fix return type for _ReadlineWrapper.get_line_buffer()

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-123228

### CODE REVIEW

**Comment by cfbolz** [comment]
implementation is fine, but why did you revert the test?

**Comment by skirpichev** [comment]
> why did you revert the test?

This wasn't tested before and I'm not sure if this does make sense.

**Comment by cfbolz** [comment]
> This wasn't tested before and I'm not sure if this does make sense.

Let's keep the test. If it had been tested before, we would have found the problem in PyPy long ago (which just shows a test is a good idea).

File: Lib/test/test_pyrepl/test_pyrepl.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import curses
```
Comment (cfbolz) [inline Lib/test/test_pyrepl/test_pyrepl.py]: this line is [failing on windows](https://github.com/python/cpython/actions/runs/10538611376/job/29201305885#step:6:625). why is the `except curses.error: pass` needed anyway?

Comment (skirpichev) [reply Lib/test/test_pyrepl/test_pyrepl.py]: Else this fails
```
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython/Lib/test/test_pyrepl/test_pyrepl.py", line 522, in test_basic
    self.assertIs(type(get_line_buffer()), str)
                       ~~~~~~~~~~~~~~~^^
  File "/home/runner/work/cpython/cpython/Lib/_pyrepl/readline.py", line 483, in get_line_buffer
    return self.get_reader().get_unicode()
           ~~~~~~~~~~~~~~~^^
  File "/home/runner/work/cpython/cpython/Lib/_pyrepl/readline.py", line 358, in get_reader
    console = Console(self.f_in, self.f_out, encoding=ENCODING)
  File "/home/runner/work/cpython/cpython/Lib/_pyrepl/unix_console.py", line 154, in __init__
    curses.setupterm(term or None, self.output_fd)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_curses.error: setupterm: could not find terminfo database`
```
(e.g. here: https://github.com/python/cpython/actions/runs/10536985217/job/29197815100)

**Comment by cfbolz** [comment]
dropping the import doesn't work, that will still make the test fail if `curses.setupterm` raises an error. but you could do this:

```python
class TestPyReplOutput(TestCase):

    ...

    def test_get_line_buffer_returns_str(self):
        reader = self.prepare_reader(code_to_events("\n"))
        wrapper = _ReadlineWrapper(reader=reader)
        self.assertIs(type(wrapper.get_line_buffer()), str)
```

which doesn't require curses at all.

**Review by cfbolz — APPROVED** [review]
looks good, I think!

**Pull request merged**